event_line.m

```diff
@@ -43,23 +43,40 @@ function event_line(temp,MHW,mclim,m90,loc,data_start,date_start,date_end,vararg
 %
 %   'Fontsize' - Default is 16. The size of font in plot.
 
-paramNames = {'Event','Color','Alpha','LineWidth','FontSize'};
-defaults   = {'MHW',[1 0.5 0.5],0.5,2,16};
+    vEvent = 'MHW';
+    vColor = [1 0.5 0.5];
+    vAlpha = 0.5;
+    vLineWidth = 2;
+    vFontSize = 16;
+    
+    paramNames = {'vEvent','vColor','vAlpha','vLineWidth','vFontSize'};
+    defaults   = {'MHW',[1 0.5 0.5],0.5,2,16};
+
+    varargin = reshape(varargin,2,length(varargin)/2);
 
-[vEvent, vColor,vAlpha,vLineWidth,vFontSize]...
-    = internal.stats.parseArgs(paramNames, defaults, varargin{:});
+    for i = 1:length(defaults)
+        if any(ismember(varargin(1,:),paramNames{i}))
+           feval(@()assignin('caller',paramNames{i},varargin{2,ismember(varargin(1,:),paramNames{i})}))
+        end      
+    end
 
-EventNames = {'MHW','MCS'};
-vEvent = internal.stats.getParamVal(vEvent,EventNames,...
-    '''Event''');
+% paramNames = {'Event','Color','Alpha','LineWidth','FontSize'};
+% defaults   = {'MHW',[1 0.5 0.5],0.5,2,16};
+% 
+% [vEvent, vColor,vAlpha,vLineWidth,vFontSize]...
+%     = internal.stats.parseArgs(paramNames, defaults, varargin{:});
+% 
+% EventNames = {'MHW','MCS'};
+% vEvent = internal.stats.getParamVal(vEvent,EventNames,...
+%     '''Event''');
 
 
 temp_here=squeeze(temp(loc(1),loc(2),:));
 
 MHW=MHW{:,:};
-MHW=MHW(MHW(:,8)==loc(1) & MHW(:,9)==loc(2),:);
```

Comment:
Adding peak_date as an output of the detect function, there is one more dimension into the table, to the numbers had to be modified accordingly